tracer/test/Datadog.Trace.Tests/Util/Http/QueryStringObfuscatorTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// <copyright file="QueryStringObfuscatorTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Util.Http;
+using FluentAssertions;
+using Moq;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Util.Http
+{
+    [Collection(nameof(QueryStringObfuscatorTests))]
+    public class QueryStringObfuscatorTests
+    {
+        private readonly TimeSpan _timeout = TimeSpan.FromSeconds(1);
+
+        [Fact]
+        public void DoesntObfuscateIfNoPattern()
+        {
+            var queryStringObfuscator = new QueryStringObfuscator.Obfuscator(_timeout);
+            var originalQueryString = "key1=val1&token=a0b21ce2-006f-4cc6-95d5-d7b550698482&key2=val2";
+            var result = queryStringObfuscator.Obfuscate(originalQueryString);
+            result.Should().Be(originalQueryString);
```

Comment:
To echo what Andrew said, I could see adding one internal static factory that looks like `Create(pattern, timeout)` (this could basically just be renamed from the `QueryStringObfuscator` constructor), which then returns the proper `ObfuscatorBase` object. That way you're testing what the product is doing when the input is an empty/invalid pattern vs when it's good

Reply:
Fixed in c5c4456c0eec985161bda650c99c0e468c5dc26b